pynats/client.py

```diff
@@ -140,8 +153,43 @@ def connect(self) -> None:
         self._socket_file = sock.makefile("rb")
         self._socket = sock
 
-        self._send_connect_command()
-        self._recv(INFO_RE)
+        scheme = self._conn_options["scheme"]
```

Comment:
I guess better to have two private methods: one for tls and one for non-tls. And use them inside `connect` method according to the settings.

For now, code looks messy: big initialization method; to my mind, we can connect to the NATS server without reconnects in case of tls; socket and socket file re-initialization; etc. Also please use separate exceptions for different situations, something like `NATSTLSConnectionRequiredError`, it's much readable and better to use from a client perspective.